extensions/amp-story/0.1/page-advancement.js

```diff
@@ -252,12 +260,18 @@ class ManualAdvancement extends AdvancementConfig {
     super();
     this.element_ = element;
     this.clickListener_ = this.maybePerformNavigation_.bind(this);
+    this.hasAutoAdvanceStr_ = this.element_.getAttribute('auto-advance-after');
+    this.isMediaElement_ = scopedQuerySelector(
+        this.element_, MEDIA_ELEMENT_SELECTOR);
   }
 
   /** @override */
   start() {
     super.start();
     this.element_.addEventListener('click', this.clickListener_, true);
+    if (!this.hasAutoAdvanceStr_ && !this.isMediaElement_) {
+      super.onProgressUpdate();
```

Comment:
Actually, isn't it reasonable to call `onProgressUpdate` unconditionally in the `start` function of all `AdvancementConfig` implementations?  I guess it would make sense to initialize the progress to something?

Reply:
@newmuis That sounds like a pretty good idea!  I'll have another look at this later in the week and make your suggested change, any more improvements you can think of please send them my way :)

Reply:
@newmuis I've merged in master and it's removed the need for checking for the onTap listener on desktop.  I could move `.onProgressUpdate()` to the base class's `start()` method, but just want to clarify the behaviour you want from the progress bar segments please.

Are you saying that you want it to look like this:

![progress-1](https://user-images.githubusercontent.com/4478671/35060904-92ac90f2-fbb7-11e7-8b58-ec25e8db47dc.gif)

Or like this:

![progress-0](https://user-images.githubusercontent.com/4478671/35060945-a17f52ea-fbb7-11e7-9df3-86fbad786f9b.gif)

Any other issues with the PR?  I'm going to give it another look over myself now having a spent a couple of days away from it.

Reply:
The second screencap is the intended behavior.  Is that what the code currently does?

Other than that, this looks good to me!  Once you confirm that the code is doing that, I can approve :smile:

Reply:
@newmuis Thank you for confirming!  Yes the current PR behaves like the 2nd screencap and everything I've tested works fine.  If I made the change you suggested it would behave like the first one, without further work done to it.

Do just want to point out the issues I've found so you can make tickets for them if they don't already exist, and are not intended UX:

1 - Videos always autoplay.
2 - `amp-audio` is currently not working with `MediaBasedAdvancement` class and gets manual advancement instead, there's a console error if you add `'amp-audio'` to `const MEDIA_ELEMENT_SELECTOR = 'amp-video';`
3 - The `MediaBasedAdvancement.fromAutoAdvanceString()` method no longer needs the `autoAdvanceStr` param, you might want to remove it from the method signature and rename the method to something more accurate

Reply:
Sounds good!

1. Intended
2. That's acceptable, at least for now (filed #12893 to track fixing)
3. Ah, I didn't catch this before.  This should actually depend on `autoAdvanceStr`, but #12604 breaks this, as the video being played is no longer the video that is specified in the DOM.  We swap out the video behind the scenes.  This can be fixed separately, but the fix will need to go in first, and then this PR will need to be updated to work accordingly.  I'll file an issue and cc you there about how we can fix this.

Thanks for the hard work on this!  Almost there :smile:

Reply:
Cool!  I'll hang tight until then 😃

Reply:
Filed #12913 to track this third line item

Reply:
Actually; I think we can do the following:

- Revert the changes to `fromAutoAdvanceString`.  This will return the auto-advance navigation to its currently-broken state (but this is the issue tracked by #12913).
- Remove the `isMediaElement_` check.  The presence of the auto-advance is enough to indicate that we should not do the initial progress update for the manual advancement mode.

Once you do that, I believe this can be approved and submitted, but the benefits for media-based advancement won't be visible until #12913 is fixed.

How does that sound?

Reply:
@newmuis Sounds pretty good mate, I'll make those changes shortly and push.